core/src/main/java/org/apache/iceberg/SerializableTable.java

```diff
@@ -247,6 +252,18 @@ public Map<String, SnapshotRef> refs() {
     return refs;
   }
 
+  @Override
+  public MetricsReporter metricsReporter() {
```

Comment:
can we add a test that makes sure the metrics reporter is preserved?

Reply:
PTAL